prog/postgres/postgres_nexus.rb

```diff
@@ -0,0 +1,184 @@
+      )
+
+      Sshable.create(
+        unix_user: "ubi",
+        host: "temp_#{vm_st.id}",
+        raw_private_key_1: ssh_key.keypair
+      ) { _1.id = vm_st.id }
+
+      server = PostgresServer.create(
+        project_id: project_id, location: location, server_name: server_name,
+        target_vm_size: vm_size, target_storage_size_gib: storage_size_gib,
+        superuser_password: SecureRandom.base64(15),
+        vm_id: vm_st.id
+      ) { _1.id = ubid.to_uuid }
+      server.associate_with_project(project)
+
+      Strand.create(prog: "Postgres::PostgresNexus", label: "start") { _1.id = server.id }
+    end
+  end
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        server.active_billing_records.each do |br|
+          br.update(span: Sequel.pg_range(br.span.begin...Time.now))
+        end
+
+        hop_destroy
+      end
+    end
+  end
+
+  label def start
+    nap 5 unless vm.strand.label == "wait"
+    vm.sshable.update(host: vm.ephemeral_net4)
+    hop_bootstrap_rhizome
+  end
+
+  label def bootstrap_rhizome
+    register_deadline(:wait, 10 * 60)
+
+    bud Prog::BootstrapRhizome, {"target_folder" => "postgres", "subject_id" => vm.id, "user" => "ubi"}
+    hop_wait_bootstrap_rhizome
+  end
+
+  label def wait_bootstrap_rhizome
+    reap
+    hop_mount_data_disk if leaf?
+    donate
+  end
+
+  label def mount_data_disk
+    case vm.sshable.cmd("common/bin/daemonizer --check format_disk")
+    when "Succeeded"
+      vm.sshable.cmd("sudo mkdir -p /dat")
+      device_path = vm.vm_storage_volumes.find { _1.boot == false }.device_path.shellescape
+
+      vm.sshable.cmd("sudo mount #{device_path} /dat")
+      hop_install_postgres
+    when "Failed", "NotStarted"
+      device_path = vm.vm_storage_volumes.find { _1.boot == false }.device_path.shellescape
+      vm.sshable.cmd("common/bin/daemonizer 'sudo mkfs --type ext4 #{device_path}' format_disk")
+    end
+
+    nap 5
+  end
+
+  label def install_postgres
+    case vm.sshable.cmd("common/bin/daemonizer --check install_postgres")
+    when "Succeeded"
+      server.incr_restart
+      hop_configure
+    when "Failed", "NotStarted"
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/install_postgres' install_postgres")
+    end
+
+    nap 5
+  end
+
+  label def configure
+    case vm.sshable.cmd("common/bin/daemonizer --check configure")
+    when "Succeeded"
+      hop_create_billing_record
+    when "Failed", "NotStarted"
+      configure_hash = server.configure_hash
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/configure' configure", stdin: JSON.generate(configure_hash))
+    end
+
+    nap 5
+  end
+
+  label def create_billing_record
+    BillingRecord.create_with_id(
+      project_id: server.project_id,
+      resource_id: server.id,
+      resource_name: server.server_name,
+      billing_rate_id: BillingRate.from_resource_properties("PostgresCores", "standard", server.location)["id"],
```

Comment:
Should we use `vm.family`?
```suggestion
      billing_rate_id: BillingRate.from_resource_properties("PostgresCores", vm.family, server.location)["id"],
```

Reply:
Not necessarily, this is the family of the postgres service, which would usually be same with vm.family, but not always. For example we might use standard VMs for a business critical postgres family. I don't want to couple vm family and postgres family.